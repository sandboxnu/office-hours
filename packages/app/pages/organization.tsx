import { Card, Col, Image, Row, Spin } from 'antd'
import { ReactElement } from 'react'
import { useProfile } from '../hooks/useProfile'
import Head from 'next/head'
import NavBar from '../components/Nav/NavBar'
import { StandardPageContainer } from '../components/common/PageContainer'
import styled from 'styled-components'
import useSWR from 'swr'
import { API } from '@koh/api-client'
import Meta from 'antd/lib/card/Meta'

const OrganizationCard = styled.div`
  margin-top: -100px;
  z-index: 2;
  background: #fff;
  border-radius: 10px;
  box-shadow: 0 1px 3px rgba(26, 26, 26, 0.1);
  @media (min-width: 320px) {
    padding: 15px;
    text-align: center;
  }
  @media (min-width: 750px) {
    padding: 10px 20px;
    text-align: left;
  }
  @media (min-width: 1000px) {
    padding: 50px 32px;
  }
  @media (min-width: 1500px) {
    max-width: 1500px;
    padding: 24px 100px;
  }
`

const CoursesSection = styled.div`
  display: flex;
  flex-wrap: wrap;
  max-width: 100%;
  margin-top: 10px;
  margin-bottom: 30px;
  gap: 15px;
`

export default function Organization(): ReactElement {
  const profile = useProfile()

  function RenderCourses(): ReactElement {
    const { data } = useSWR(`/api/v1/courses`, async () =>
      API.course.getOrganizationCourses(profile?.organization.id),
    )

    return data ? (
      <>
        <CoursesSection>
          {data?.coursesPartial.map((course) => (
            <Card
              key={course.id}
              style={{
                boxShadow: '0 4px 8px 0 rgba(0,0,0,0.2)',
              }}
              cover={
                <img
                  alt="example"
                  style={{ objectFit: 'cover', height: '15vh' }}
                  src="https://open-2021.sites.olt.ubc.ca/files/2020/10/OSIP-2020-Slider.jpg"
                />
              }
            >
              <div
                style={{
                  display: 'flex',
                  verticalAlign: 'middle',
                  alignItems: 'center',
                  justifyContent: 'space-between',
                }}
              >
                <Meta title={course.name} />
              </div>
            </Card>
          ))}
        </CoursesSection>
      </>
    ) : (
      <></>
    )
  }

  return profile ? (
    <>
      <Head>
        <title>{profile?.organization.organizationName}</title>
      </Head>
      <StandardPageContainer style={{ flex: 'none' }}>
        <NavBar />
      </StandardPageContainer>
      <Image
        preview={false}
        src={`/api/v1/organization/${profile?.organization.id}/get_banner/${profile?.organization.organizationBannerUrl}`}
        style={{
          width: '100%',
          height: '40vh',
          objectFit: 'cover',
          objectPosition: 'center',
        }}
      />

      <StandardPageContainer>
        <OrganizationCard>
          <Row
            gutter={{ xs: 8, sm: 16, md: 24, lg: 32 }}
            style={{ alignItems: 'center' }}
          >
            <Col xs={{ span: 24 }} sm={{ span: 3 }}>
              <Image
                preview={false}
                src={`/api/v1/organization/${profile?.organization.id}/get_logo/${profile?.organization.organizationLogoUrl}`}
                style={{
                  width: '100%',
                  height: '10vh',
                  objectFit: 'contain',
                  objectPosition: 'center',
                }}
              />
            </Col>
            <Col xs={{ span: 24 }} sm={{ span: 21 }}>
              <h1>{profile?.organization.organizationName}</h1>
              <p>{profile?.organization.organizationDescription}</p>
            </Col>
          </Row>
        </OrganizationCard>

        <div style={{ marginTop: 20 }}>
          <h2>Courses Available</h2>
          <RenderCourses />
        </div>
      </StandardPageContainer>
    </>
  ) : (
    <Spin />
  )
}
