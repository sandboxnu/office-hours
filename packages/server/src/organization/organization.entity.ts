import { Exclude } from 'class-transformer';
import {
  BaseEntity,
  Column,
  Entity,
  JoinColumn,
  OneToMany,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { OrganizationUserModel } from './organization-user.entity';
import { OrganizationCourseModel } from './organization-course.entity';

@Entity('organization_model')
export class OrganizationModel extends BaseEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column('text')
  name: string;

  @Column('text', { nullable: true })
  description: string;

  @Column('text', { nullable: true })
  logoUrl: string;

  @Column('text', { nullable: true })
  bannerUrl: string;

  @Column('text', { nullable: true })
  websiteUrl: string;

  @Column('boolean', { default: false })
  ssoEnabled: boolean;

  @Column('text', { nullable: true })
  ssoUrl: string;

  @Exclude()
  @JoinColumn({ name: 'organizationId' })
  @OneToMany(
    type => OrganizationUserModel,
    organizationUser => organizationUser.organization,
  )
  organizationUsers: OrganizationUserModel[];

  @Exclude()
  @JoinColumn({ name: 'organizationId' })
  @OneToMany(
    type => OrganizationCourseModel,
    organizationCourse => organizationCourse.organization,
  )
  organizationCourses: OrganizationCourseModel[];
}
