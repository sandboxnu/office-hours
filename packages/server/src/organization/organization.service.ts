import { Injectable, NotFoundException } from '@nestjs/common';
import { OrganizationUserModel } from './organization-user.entity';
import { UserModel } from 'profile/user.entity';
import { Brackets, getRepository } from 'typeorm';
import { OrganizationCourseModel } from './organization-course.entity';
import { CourseModel } from 'course/course.entity';
import { GetOrganizationUserResponse, Role, UserRole } from '@koh/common';
import { UserCourseModel } from 'profile/user-course.entity';

export interface UserResponse {
  userId: number;
  firstName: string;
  lastName: string;
  email: string;
  photoUrl: string | null;
  userRole: string;
  organizationRole: string;
}

export interface FlattenedOrganizationResponse {
  id: number;
  organizationName: string;
  organizationDescription: string;
  organizationLogoUrl: string;
  organizationBannerUrl: string;
  organizationRole: string;
}

export interface OrganizationCourseResponse {
  id: number;
  organizationId: number;
  courseId: number;
  course: CourseModel;
  profIds: Array<number>;
}

export interface CourseResponse {
  courseId: number;
  courseName: string;
}

@Injectable()
export class OrganizationService {
  public async getOrganizationRoleByUserId(userId: number): Promise<string> {
    const organizationUser = await OrganizationUserModel.findOne({
      where: {
        userId,
      },
    });

    if (!organizationUser) {
      return null;
    }
    return organizationUser.role;
  }

  public async deleteUserCourses(
    userId: number,
    userCourses: number[],
  ): Promise<void> {
    const user = await UserModel.findOne({
      where: {
        id: userId,
      },
    });

    if (!user) {
      throw new NotFoundException(`User with id ${userId} not found`);
    }

    for (const courseId of userCourses) {
      const course = await CourseModel.findOne({
        where: {
          id: courseId,
        },
      });

      if (!course) {
        throw new NotFoundException(`Course with id ${courseId} not found`);
      }

      await UserCourseModel.delete({
        userId: user.id,
        courseId: course.id,
      });
    }
  }

  public async getCourses(
    organizationId: number,
    page: number,
    pageSize: number,
    search?: string,
  ): Promise<CourseResponse[]> {
    const organizationCourses = await getRepository(OrganizationCourseModel)
      .createQueryBuilder()
      .leftJoin(
        CourseModel,
        'CourseModel',
        'CourseModel.id = OrganizationCourseModel.courseId',
      )
      .where('OrganizationCourseModel.organizationId = :organizationId', {
        organizationId,
      });

    if (search) {
      const likeSearch = `%${search.replace(' ', '')}%`.toUpperCase();
      organizationCourses.andWhere(
        new Brackets((q) => {
          q.where('UPPER("CourseModel"."name") like :searchString', {
            searchString: likeSearch,
          });
        }),
      );
    }

    const courses = organizationCourses.select([
      'CourseModel.id as courseId',
      'CourseModel.name as courseName',
    ]);

    const coursesSubset = await courses
      .orderBy('CourseModel.name')
      .skip((page - 1) * pageSize)
      .take(pageSize)
      // .getMany() wouldn't work here because relations are not working well with getMany()
      .getRawMany();

    const coursesResponse = coursesSubset.map((course) => {
      return {
        courseId: course.courseid,
        courseName: course.coursename,
      };
    });

    return coursesResponse;
  }

  public async getUsers(
    organizationId: number,
    page: number,
    pageSize: number,
    search?: string,
  ): Promise<UserResponse[]> {
    const organizationUsers = await getRepository(OrganizationUserModel)
      .createQueryBuilder()
      .leftJoin(
        UserModel,
        'UserModel',
        'UserModel.id = OrganizationUserModel.userId',
      )
      .where('OrganizationUserModel.organizationId = :organizationId', {
        organizationId,
      });

    if (search) {
      const likeSearch = `%${search.replace(' ', '')}%`.toUpperCase();
      organizationUsers.andWhere(
        new Brackets((q) => {
          q.where(
            'CONCAT(UPPER("UserModel"."firstName"), UPPER("UserModel"."lastName")) like :searchString',
            {
              searchString: likeSearch,
            },
          );
        }),
      );
    }

    const users = organizationUsers.select([
      'UserModel.id as userId',
      'UserModel.firstName as userFirstName',
      'UserModel.lastName as userLastName',
      'UserModel.email as userEmail',
      'UserModel.photoURL as userPhotoUrl',
      'UserModel.userRole as userRole',
      'OrganizationUserModel.role as userOrganizationRole',
    ]);

    const usersSubset = await users
      .orderBy('UserModel.lastName')
      .skip((page - 1) * pageSize)
      .take(pageSize)
      // .getMany() wouldn't work here because relations are not working well with getMany()
      .getRawMany();

    const usersResponse = usersSubset.map((user) => {
      return {
        userId: user.userid,
        firstName: user.userfirstname,
        lastName: user.userlastname,
        email: user.useremail,
        photoUrl: user.userphotourl,
        userRole: user.userrole,
        organizationRole: user.userorganizationrole,
      };
    });

    return usersResponse;
  }

  public async getOrganizationUserByUserId(
    userId: number,
  ): Promise<GetOrganizationUserResponse> {
    const organizationUserResponse = await OrganizationUserModel.findOne({
      where: {
        userId,
      },
      relations: [
        'organizationUser',
        'organizationUser.courses',
        'organizationUser.courses.course',
      ],
    });

    if (!organizationUserResponse) {
      throw new NotFoundException(
        `OrganizationUser with userId ${userId} not found`,
      );
    }

    const { organizationId, role, organizationUser } = organizationUserResponse;

    const globalRole: string =
      organizationUser.userRole == UserRole.ADMIN ? 'unknown' : 'user';

    const flattenedUser = {
      organizationId: organizationId,
      organizationRole: role,
      user: {
        id: organizationUser.id,
        firstName: organizationUser.firstName,
        lastName: organizationUser.lastName,
        email: organizationUser.email,
        photoUrl: organizationUser.photoURL,
        fullName: organizationUser.name,
        globalRole: globalRole,
        sid: organizationUser.sid,
        accountDeactivated: organizationUser.accountDeactivated,
      },
      courses: organizationUser.courses.map((courseInfo) => {
        const { role, course } = courseInfo;

        return {
          id: course.id,
          name: course.name,
          role: role,
        };
      }),
    };

    return flattenedUser;
  }

  public async getOrganizationCourse(
    organizationId: number,
    courseId: number,
  ): Promise<OrganizationCourseResponse> {
    const organizationCourse = await OrganizationCourseModel.findOne({
      where: {
        courseId,
        organizationId,
      },
      relations: ['course'],
    });

    if (!organizationCourse) {
      throw new NotFoundException(
        `OrganizationCourse with organizationId ${organizationId} and courseId ${courseId} not found`,
      );
    }

    const professors = await UserCourseModel.find({
      where: {
        courseId,
        role: Role.PROFESSOR,
      },
    });

    let profIds = [];

    if (professors) {
      profIds = professors.map((professor) => professor.userId);
    }

    return {
      ...organizationCourse,
      profIds,
    };
  }

  public async getOrganizationAndRoleByUserId(
    userId: number,
  ): Promise<FlattenedOrganizationResponse> {
    const organizationUser = await OrganizationUserModel.createQueryBuilder(
      'organizationUser',
    )
      .leftJoinAndSelect('organizationUser.organization', 'organization')
      .where('organizationUser.userId = :userId', { userId })
      .getOne();

    if (!organizationUser) {
      return null;
    }

    const flattenedOrganization = {
      id: organizationUser.organizationId,
      organizationName: organizationUser.organization.name,
      organizationDescription: organizationUser.organization.description,
      organizationLogoUrl: organizationUser.organization.logoUrl,
      organizationBannerUrl: organizationUser.organization.bannerUrl,
      organizationRole: organizationUser.role,
    };

    return flattenedOrganization;
  }
}
