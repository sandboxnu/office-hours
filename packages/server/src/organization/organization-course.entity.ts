import { Exclude } from 'class-transformer';
import {
  BaseEntity,
  Column,
  Entity,
  ManyToOne,
  OneToMany,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { OrganizationModel } from './organization.entity';
import { CourseModel } from '../course/course.entity';

@Entity('organization_course_model')
export class OrganizationCourseModel extends BaseEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Exclude()
  @ManyToOne(
    type => OrganizationModel,
    organization => organization.organizationCourses,
  )
  organization: OrganizationModel;

  @Column({ nullable: true })
  organizationId: number;

  @Exclude()
  @OneToMany(
    type => CourseModel,
    course => course.organizationCourse,
    {
      cascade: true,
    },
  )
  course: CourseModel;

  @Column({ nullable: true })
  courseId: number;
}
