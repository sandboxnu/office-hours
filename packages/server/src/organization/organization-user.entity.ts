import {
  BaseEntity,
  Column,
  Entity,
  ManyToOne,
  OneToOne,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { OrganizationModel } from './organization.entity';
import { Exclude } from 'class-transformer';
import { UserModel } from '../profile/user.entity';
import { OrganizationRole } from '@koh/common';

@Entity('organization_user_model')
export class OrganizationUserModel extends BaseEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Exclude()
  @ManyToOne(
    type => OrganizationModel,
    organization => organization.organizationUsers,
  )
  organization: OrganizationModel;

  @Exclude()
  @OneToOne(
    type => UserModel,
    user => user.organizationUser,
  )
  organizationUser: UserModel;

  @Column({ nullable: true })
  userId: number;

  @Column({ nullable: true })
  organizationId: number;

  @Column({
    type: 'enum',
    enum: OrganizationRole,
    default: OrganizationRole.MEMBER,
  })
  role: OrganizationRole;
}
