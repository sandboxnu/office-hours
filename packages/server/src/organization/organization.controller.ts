/* eslint-disable @typescript-eslint/no-unused-vars */
import {
  Controller,
  Get,
  HttpException,
  HttpStatus,
  Param,
  Post,
  Res,
  UseGuards,
} from '@nestjs/common';
import { UserModel } from 'profile/user.entity';
import { Response } from 'express';
import { ERROR_MESSAGES, OrganizationRole } from '@koh/common';
import { OrganizationUserModel } from './organization-user.entity';
import { JwtAuthGuard } from 'guards/jwt-auth.guard';
import { CourseModel } from 'course/course.entity';
import { OrganizationCourseModel } from './organization-course.entity';
import { OrganizationModel } from './organization.entity';

@Controller('organization')
export class OrganizationController {
  @Post(':oid/add_course/:cid')
  @UseGuards(JwtAuthGuard)
  async addCourseToOrganization(
    @Res() res: Response,
    @Param('oid') oid: string,
    @Param('cid') cid: string,
  ): Promise<void> {
    CourseModel.findOne({
      where: { id: cid },
    })
      .then((course) => {
        if (!course) {
          throw new HttpException(
            ERROR_MESSAGES.courseController.courseNotFound,
            HttpStatus.NOT_FOUND,
          );
        }

        OrganizationCourseModel.findOne({
          where: { courseId: cid, organizationId: oid },
        })
          .then((organizationCourse) => {
            if (organizationCourse) {
              throw new HttpException(
                ERROR_MESSAGES.organizationController.courseAlreadyInOrganization,
                HttpStatus.BAD_REQUEST,
              );
            }

            const organizationCourseModel = new OrganizationCourseModel();
            organizationCourseModel.courseId = parseInt(cid);
            organizationCourseModel.organizationId = parseInt(oid);

            organizationCourseModel
              .save()
              .then((_) => {
                res
                  .status(200)
                  .send({ message: 'Course added to organization' });
              })
              .catch((err) => {
                res.status(500).send({ message: err });
              });
          })
          .catch((err) => {
            res.status(500).send({ message: err });
          });
      })
      .catch((err) => {
        res.status(500).send({ message: err });
      });
  }

  @Get(':oid')
  @UseGuards(JwtAuthGuard)
  async get(@Res() res: Response, @Param('oid') oid: string): Promise<void> {
    OrganizationModel.findOne({
      where: { id: oid },
    })
      .then((organization) => {
        if (!organization) {
          return res.status(HttpStatus.NOT_FOUND).send({
            message: ERROR_MESSAGES.organizationController.organizationNotFound,
          });
        }

        res.status(HttpStatus.OK).send(organization);
      })
      .catch((err) => {
        res.status(500).send({ message: err });
      });
  }

  @Post(':oid/add_member/:uid')
  @UseGuards(JwtAuthGuard)
  async addUserToOrganization(
    @Res() res: Response,
    @Param('oid') oid: string,
    @Param('uid') uid: string,
  ): Promise<void> {
    UserModel.findOne({
      where: { id: uid },
    })
      .then((user) => {
        if (!user) {
          throw new HttpException(
            ERROR_MESSAGES.profileController.accountNotAvailable,
            HttpStatus.NOT_FOUND,
          );
        }

        OrganizationUserModel.findOne({
          where: { userId: uid, organizationId: oid },
        })
          .then((organizationUser) => {
            if (organizationUser) {
              throw new HttpException(
                ERROR_MESSAGES.organizationController.userAlreadyInOrganization,
                HttpStatus.BAD_REQUEST,
              );
            }

            const organizationUserModel = new OrganizationUserModel();
            organizationUserModel.organizationId = parseInt(oid);
            organizationUserModel.userId = parseInt(uid);
            organizationUserModel.role = OrganizationRole.MEMBER;

            organizationUserModel
              .save()
              .then((_) => {
                res.status(200).send({ message: 'User added to organization' });
              })
              .catch((err) => {
                res.status(500).send({ message: err });
              });
          })
          .catch((err) => {
            res.status(500).send({ message: err });
          });
      })
      .catch((err) => {
        res.status(500).send({ message: err });
      });
  }
}
