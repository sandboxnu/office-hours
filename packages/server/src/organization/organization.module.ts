import { Module } from '@nestjs/common';
import { OrganizationService } from './organization.service';
import { OrganizationController } from './organization.controller';

@Module({
  imports: [],
  controllers: [OrganizationController],
  providers: [OrganizationService],
  exports: [OrganizationService],
})
export class OrganizationModule {}
