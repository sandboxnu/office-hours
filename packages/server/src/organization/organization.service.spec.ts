import { Connection } from 'typeorm';
import { OrganizationService } from './organization.service';
import { Test, TestingModule } from '@nestjs/testing';
import { TestConfigModule, TestTypeOrmModule } from '../../test/util/testUtils';
import { OrganizationUserModel } from './organization-user.entity';
import {
  CourseFactory,
  OrganizationFactory,
  UserFactory,
} from '../../test/util/factories';
import { OrganizationRole } from '@koh/common';
import { OrganizationCourseModel } from './organization-course.entity';

describe('OrganizationService', () => {
  let service: OrganizationService;
  let conn: Connection;

  beforeAll(async () => {
    const module: TestingModule = await Test.createTestingModule({
      imports: [TestTypeOrmModule, TestConfigModule],
      providers: [OrganizationService],
    }).compile();
    service = module.get<OrganizationService>(OrganizationService);
    conn = module.get<Connection>(Connection);
  });

  afterAll(async () => {
    await conn.close();
  });

  beforeEach(async () => {
    await conn.synchronize(true);
  });

  describe('getOrganizationRoleByUserId', () => {
    it('should return null if no organizationUser exists', async () => {
      const role = await service.getOrganizationRoleByUserId(0);
      expect(role).toBeNull();
    });

    it('should return the role of the organizationUser', async () => {
      const user = await UserFactory.create();
      const organization = await OrganizationFactory.create();

      await OrganizationUserModel.create({
        userId: user.id,
        organizationId: organization.id,
      }).save();

      const role = await service.getOrganizationRoleByUserId(user.id);
      expect(role).toEqual(OrganizationRole.MEMBER);
    });
  });

  describe('getOrganizationAndRoleByUserId', () => {
    it('should return null if no organizationUser exists', async () => {
      const role = await service.getOrganizationAndRoleByUserId(0);
      expect(role).toBeNull();
    });

    it('should return the organization and role of the organizationUser', async () => {
      const user = await UserFactory.create();
      const organization = await OrganizationFactory.create();

      await OrganizationUserModel.create({
        userId: user.id,
        organizationId: organization.id,
      }).save();

      const organizationUserModel =
        await service.getOrganizationAndRoleByUserId(user.id);
      expect(organizationUserModel).toMatchSnapshot();
    });
  });

  describe('getCourses', () => {
    it('should return organization courses', async () => {
      const organization = await OrganizationFactory.create();
      const course = await CourseFactory.create();

      await OrganizationCourseModel.create({
        organizationId: organization.id,
        courseId: course.id,
      }).save();

      const courses = await service.getCourses(organization.id, 1, 50);
      expect(courses).toMatchSnapshot();
    });
  });

  describe('getUsers', () => {
    it('should return organization users', async () => {
      const organization = await OrganizationFactory.create();
      const userOne = await UserFactory.create();
      const userTwo = await UserFactory.create();

      await OrganizationUserModel.create({
        organizationId: organization.id,
        userId: userOne.id,
      }).save();

      await OrganizationUserModel.create({
        organizationId: organization.id,
        userId: userTwo.id,
      }).save();

      const users = await service.getUsers(organization.id, 1, 50);
      expect(users).toMatchSnapshot();
    });
  });
});
