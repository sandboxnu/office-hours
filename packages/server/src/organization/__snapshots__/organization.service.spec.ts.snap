// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationService getCourses should return organization courses 1`] = `
Array [
  Object {
    "courseId": 1,
    "courseName": "CS 304",
  },
]
`;

exports[`OrganizationService getOrganizationAndRoleByUserId should return the organization and role of the organizationUser 1`] = `
Object {
  "id": 1,
  "organizationDescription": "UBC Okanagan",
  "organizationLogoUrl": null,
  "organizationName": "UBCO",
  "organizationRole": "member",
}
`;

exports[`OrganizationService getUsers should return organization users 1`] = `
Array [
  Object {
    "email": "user@ubc.ca",
    "firstName": "User",
    "lastName": "Person",
    "organizationRole": "member",
    "photoUrl": null,
    "userId": 1,
    "userRole": "user",
  },
  Object {
    "email": "user@ubc.ca",
    "firstName": "User",
    "lastName": "Person",
    "organizationRole": "member",
    "photoUrl": null,
    "userId": 2,
    "userRole": "user",
  },
]
`;
